guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,750 @@
+
+Luckily, with KerasCV this is easy:
+"""
+
+best_decoder = None
+score_to_beat = 0
+worst_score = 1.0
+
+iou_thresholds = [0.35, 0.5, 0.65]
+confidence_thresholds = [0.5, 0.75, 0.9]
+for iou_threshold in tqdm(iou_thresholds):
+    for confidence_threshold in confidence_thresholds:
+        coco_metrics.reset_state()
+        prediction_decoder = keras_cv.layers.MultiClassNonMaxSuppression(
+            bounding_box_format="xywh",
+            from_logits=True,
+            # Decrease the required threshold to make predictions get pruned out
+            iou_threshold=iou_threshold,
+            # Tune confidence threshold for predictions to pass NMS
+            confidence_threshold=iou_threshold,
+        )
+        model.prediction_decoder = prediction_decoder
+
+        # Remove take(20) in a production setting
+        coco_metrics.reset_state()
+        model.evaluate(eval_ds.take(20))
+        result = coco_metrics.result(force=True)
+        if result["MaP"] > score_to_beat:
+            best_decoder = prediction_decoder
+            score_to_beat = result["MaP"]
+
+        if result["MaP"] < worst_score:
+            worst_score = result["MaP"]
+
+model.prediction_decoder = best_decoder
+print(
+    f"Best scores found with iou_threshold={best_decoder.iou_threshold},
+    f"confidence_threshold={best_decoder.confidence_threshold}. Best MaP is "
+    f"{score_to_beat}, worst MaP is {worst_score}."
+)
+
+"""
+Lets visualize the results using our optimal decoder:
+"""
+
+visualize_detections(model, bounding_box_format="xywh", dataset=eval_ds.shuffle(8))
+
+"""
+Awesome!
+
+One final helpful pattern to be aware of is monitoring training is to visualize
+detections in a `keras.callbacks.Callback`:
+"""
+
+
+class VisualizeDetections(keras.callbacks.Callback):
+    def on_epoch_end(self, epoch, logs):
+        visualize_detections(self.model, eval_ds, bounding_box_format="xywh")
+
+
+"""
+## Takeaways and Next Steps
+
+KerasCV makes it easy to construct state-of-the-art object detection pipelines.
+In this guide, we started off by writing a data loader using the KerasCV
+bounding box specification.
+Following this, we assembled a production grade data augmentation pipeline using
+the module `KerasCV` preprocessing layers in <50 lines of code.
+We constructed a RetinaNet and trained for an epoch.
+
+KerasCV object detection components can be used independently, but also have deep
+integration with each other.
+KerasCV makes authoring production grade bounding box augmentation,
+model training, visualization, and
+metric evaluation easy.
+
+Some follow up exercises for the reader:
+
+- add additional augmentation techniques to improve model performance
+- tune the hyperparameters and data augmentation used to produce high quality results
+- train an object detection model on your own dataset
+
+One last fun code snippet to showcase the power of KerasCV's modular API!
```

Comment:
I'm not sure this showcases modularity. We're passing the outputs of one monolithic API to another. It does showcase the diversity of our offerings however.

Reply:
removed -modularity